src/model.jl

```diff
@@ -263,11 +299,19 @@ function _add_constraint_load_shed!(
         bus_demand[sets.load_bus_idx[i], j],
     )
     if demand_flexibility.enabled
-        demand_for_load_shed = JuMP.@expression(
-            m,
-            [i=1:sets.num_load_bus, j=1:interval_length],
-            demand_for_load_shed[i, j] + m[:load_shift_up][i, j] - m[:load_shift_dn][i, j],
-        )
+        if demand.flexibility.cost_input_type == "AREA"
+            demand_for_load_shed = JuMP.@expression(
+                m,
+                [i=1:sets.num_load_bus, j=1:interval_length],
+                demand_for_load_shed[i, j] + m[:load_shift_up][i, j] - m[:load_shift_dn][i, j],
+            )
+        elseif demand.flexibility.cost_input_type == "BUS"
+            demand_for_load_shed = JuMP.@expression(
+                m,
+                [i=1:sets.num_load_bus, j=1:interval_length],
+                demand_for_load_shed[i, j] + m[:load_shift_up][i, j] - m[:load_shift_dn][i, j],
```

Comment:
Do we need a mapping here between the set of load buses and the set of demand flexibility buses? Unless the sets are equal (i.e., the `AREA` configuration), I believe `i` will exceed the index range for `load_shift_up` and `load_shift_dn` (and the buses would probably be somewhat misaligned, but exceeding the index range would likely be the thing that throws the error).

Reply:
right this seems to be a mistake. the mapping used in _add_constraint_power_balance() should be copied here as well. hopefully it should be moved to _make_sets() so it only needs to be computed 1 time.